src/lib/reticket.js

```diff
@@ -143,26 +148,44 @@ export async function reticketPointBetweenWallets({
   progress(TRANSACTION_PROGRESS.SIGNING);
 
   const suggestedGasPrice = await getSuggestedGasPrice(networkType);
+  const chainId = await web3.eth.net.getId();
   const gasPrice = toWei(suggestedGasPrice.toFixed(), 'gwei');
+  console.log(suggestedGasPrice.toFixed());
   const gasPriceBN = toBN(gasPrice);
-  let totalCost = toBN(0);
+  console.log(gasPrice);
   let inviteNonce = await web3.eth.getTransactionCount(fromWallet.address);
-  txs = txs.map(tx => {
-    tx.from = fromWallet.address;
-    tx.nonce = inviteNonce++;
-    tx.gasPrice = gasPriceBN;
-    totalCost = totalCost.add(gasPriceBN.mul(toBN(tx.gas)));
-    return tx;
-  });
+  console.log(txs);
+  const totalCost = txs.reduce(
+    (acc, tx) => acc.add(gasPriceBN.mul(toBN(tx.gas))),
+    toBN(0)
+  );
 
-  let txPairs = txs.map(tx => {
-    let stx = new Tx(tx);
-    stx.sign(Buffer.from(fromWallet.privateKey, 'hex'));
-    return {
-      raw: hexify(stx.serialize()),
-      signed: stx,
-    };
-  });
+  console.log(txs);
+
+  let txPairs = await Promise.all(
+    txs.map(async (tx, idx) => {
+      const stx = await signTransaction({
```

Comment:
I suppose the UX for this in the previously-not-supported cases is just repeated pop-ups/prompts to do the signing?

Reply:
Perhaps this is too eager in sending transactions to the device, and we want to actually run this in sequence? ie, maybe not rely on `.map` here? Not sure, but seems like something worth trying if we don't have any other leads.

Reply:
thanks for the heads up. yeah the popups/prompts just repeat